tests/test-spawn-snid.lua

```diff
@@ -0,0 +1,44 @@
+--- Tests for spawn's startup notifications.
+
+local spawn = require("awful.spawn")
+
+local manage_called, c_snid
+
+client.connect_signal("manage", function(c)
+  manage_called = true
+  c_snid = c.startup_id
+  print("manage_called: c_snid:", tostring(c_snid))
```

Comment:
Hm. Does the urxvt version on travis actually support startup notification? Looking at their `configure.ac`, it's an optional feature.
Ideas:
* Try without my patches so that we can agree that it's not my fault. :-)
* Try something like gnome-terminal, gedit, kate, ... (something which "surely" supports startup notification)
* These travis/cmake-fixes should be merged to master soonish, so that other branches can be rebased. This is independent from the rest of this PR (and I don't think we care much for SSL-certificates, because it's "only" a temporary travis machine that could be attacked).